jobbergate-api/jobbergate_api/apps/job_submissions/routers.py

```diff
@@ -186,6 +207,85 @@ async def job_script_update(job_submission_id: int, job_submission: JobSubmissio
     return job_submission_data
 
 
+# The "agent" routes are used for agents to fetch pending job submissions and update their statuses
+@router.get(
+    "/job-submissions/agent/pending",
+    description="Endpoint to list pending job_submissions for the requesting client",
+    response_model=List[PendingJobSubmission],
+)
+async def job_submissions_agent_pending(
+    token_payload: TokenPayload = Depends(guard.lockdown(Permissions.JOB_SUBMISSIONS_VIEW)),
+):
+    """
+    Get a list of pending job submissions for the cluster-agent.
+    """
+    if token_payload.client_id is None:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="Access token does not contain a `client_id`. Cannot fetch pending submissions",
+        )
+
+    query = (
+        select(
+            columns=[
+                job_submissions_table.c.id,
+                job_submissions_table.c.job_submission_name,
+                job_scripts_table.c.job_script_name,
+                job_scripts_table.c.job_script_data_as_string,
+                applications_table.c.application_name,
+            ]
+        )
```

Comment:
I'm wondering if we could since select the rows that don't have the `slurm_job_id` set because this parameter should be set only when the agent updates it. It seems a more efficient implementation for me

Reply:
If you allow an enum column to be nullable, you essentially add an additional implict value to the enum (null). Then you have an issue where trying to understand the status of a specific row means you have to understand the implementation fully (what does null mean?).

I prefer to make enum columns non-nullable and for the rows to have an explicit status.